include/seqan3/alphabet/nucleotide/dna4.hpp

```diff
@@ -269,9 +287,120 @@ constexpr dna4 dna4::T{internal_type::T};
 constexpr dna4 dna4::U{dna4::T};
 constexpr dna4 dna4::UNKNOWN{dna4::A};
 
+} // namespace seqan3
+
+namespace seqan3::detail
+{
+
+//!\brief seqan3::dna4 is defined as being a nucleotide alphabet.
+template <>
+struct is_nucleotide<dna4> : public std::true_type
+{};
+
+} // namespace seqan3::detail
+
 #ifndef NDEBUG
-static_assert(alphabet_concept<dna4>);
-static_assert(detail::internal_alphabet_concept<dna4>);
+static_assert(seqan3::alphabet_concept<seqan3::dna4>);
+static_assert(seqan3::nucleotide_concept<seqan3::dna4>);
 #endif
 
+// ------------------------------------------------------------------
+// containers
+// ------------------------------------------------------------------
+
+namespace seqan3
+{
+
+//!\brief Alias for an std::vector of seqan3::dna4.
+//!\relates dna4
+using dna4_vector = std::vector<dna4>;
+
+
+/*!\brief Alias for an std::basic_string of seqan3::dna4.
+ * \relates dna4
+ *
+ * \attention
+ * Note that we recommend using @link dna4_vector @endlink instead of dna4_string in almost all situations.
+ * While the C++ style operations on the string are well supported, you should not access the internal c-string
+ * and should not use C-Style operations on it, e.g. the `char_traits::strlen` function will not return the
+ * correct length of the string (while the `.size()` returns the correct value).
+ */
+using dna4_string = std::basic_string<dna4, std::char_traits<dna4>>;
+
 } // namespace seqan3
+
+// ------------------------------------------------------------------
+// literals
+// ------------------------------------------------------------------
+
+namespace seqan3::literal
+{
+
+/*!\brief dna4 literal (returns @link dna4_vector @endlink)
+ * \relates seqan3::dna4
+ *
+ * You can use this string literal to easily assign to dna4_vector:
+ *
+ *~~~~~~~~~~~~~~~{.cpp}
+ *     // these don't work:
+ *     // dna4_vector foo{"ACGTTA"};
+ *     // dna4_vector bar = "ACGTTA";
+ *
+ *     // but these do:
+ *     using namespace seqan3::literal;
+ *     dna4_vector foo{"ACGTTA"_dna4};
+ *     dna4_vector bar = "ACGTTA"_dna4;
+ *     auto bax = "ACGTTA"_dna4;
+ *~~~~~~~~~~~~~~~
+ *
+ * \attention
+ * All user-defined literals are in the namespace seqan3::literal!
```

Comment:
I am currently unhappy about the `view` namespace anyway. Maybe we should discuss with @rrahn whether we get rid of it altogether? I will change the wording in any case.